### ISSUE TITLE
data rule for bottlefield that sometimes has data

### ISSUE THREAD

**Issue by adamstauffer** [dunbarcyber/cyphon#374]
I have a container that is configured like the following:

```
{
    "@timestamp": "DateTimeField (DateTime)",
    "agent": {
        "hostname": "CharField",
        "id": "CharField",
        "name": "CharField (Keyword)"
    },
    "decoder": {
        "name": "CharField (Keyword)"
    },
    "file": "CharField",
    "full_log": "CharField",
    "host": "CharField",
    "input_type": "CharField",
    "location": "CharField (Location)",
    "log": "CharField",
    "manager": {
        "name": "CharField (Keyword)"
    },
    "rule": {
        "cis": "CharField",
        "description": "TextField (Keyword)",
        "firedtimes": "IntegerField",
        "groups": "CharField",
        "id": "CharField",
        "level": "CharField"
    },
    "source": "CharField",
    "timestamp": "CharField",
    "title": "CharField",
    "type": "CharField"
}
```

the issue im having is that i get data into elasticsearch that sometimes has a `rule` configured, but sometimes not. like so:

```
{
    ...
    "input_type": "log",
    "rule": {},
    "decoder": {
        "name": "rootcheck"
    },
    ...
    "timestamp": "2018-02-01T11: 37: 09-0500"
}
```

however, i need a datarule that detects when the `rule.level` breaches a certain threshold. if i configure this datarule, i get this traceback from the watchdog when it tries to handle the above message:

```
Traceback (most recent call last):
  File "receiver/receiver.py", line 120, in process_msg
    consumer_func(doc_obj)
  File "/usr/src/app/cyphon/cyphon/transaction.py", line 97, in _decorator
    result = func(*args, **kwargs)
  File "/usr/src/app/cyphon/alarms/models.py", line 72, in process
    alarm.process(doc_obj)
  File "/usr/src/app/cyphon/watchdogs/models.py", line 200, in process
    alert_level = self.inspect(doc_obj.data)
  File "/usr/src/app/cyphon/watchdogs/models.py", line 179, in inspect
    if trigger.is_match(data):
  File "/usr/src/app/cyphon/watchdogs/models.py", line 316, in is_match
    return self.sieve.is_match(data)
  File "/usr/src/app/cyphon/sifter/sieves/models.py", line 507, in is_match
    match = self._matches_all(data)
  File "/usr/src/app/cyphon/sifter/sieves/models.py", line 479, in _matches_all
    if not node.is_match(data):
  File "/usr/src/app/cyphon/sifter/sieves/models.py", line 597, in is_match
    return self.node_object.is_match(data)
  File "/usr/src/app/cyphon/sifter/sieves/models.py", line 232, in is_match
    match = self._check_value(value)
  File "/usr/src/app/cyphon/sifter/sieves/models.py", line 397, in _check_value
    return func(value)
  File "/usr/src/app/cyphon/sifter/sieves/models.py", line 380, in _numeric_match
    return operators[comparison](float(value), float(self.value))
TypeError: float() argument must be a string or a number, not 'NoneType'
```

do you have any recommendations for a way to have a bottlefield that sometimes has data and only try to run datarules on it if they aren't null? or does this break the concept of the "bottle" in that my data can have variations? or should cyphon do a better job of handling this case? i could probably submit a PR to do the latter but wanted to make sure there wasnt some obvious configuration i could change

**Comment by lhadjchikh** [comment]
I think you're right; it should just do a better job of handling this case. Please feel free to submit a PR. Thanks!

**Comment by lhadjchikh** [comment]
@adamstauffer Should be fixed in the `dunbar/cyphon:latest` Docker image. Let me know if it still gives any problems, and thanks for the bug report/fix!

**Comment by lhadjchikh** [comment]
Patched in [1.6.1](https://github.com/dunbarcyber/cyphon/releases/tag/1.6.1). Thanks, @adamstauffer!
